src/block/decompress.rs

```diff
@@ -191,14 +191,14 @@ pub fn decompress_into_with_dict(
     output_pos: usize,
     ext_dict: &[u8],
 ) -> Result<usize, DecompressError> {
-    decompress_internal::<true>(input, &mut (output, output_pos).into(), ext_dict)
+    decompress_internal::<true>(input, &mut Sink::new(output, output_pos), ext_dict)
 }
 
 /// Decompress all bytes of `input` into `output`.
 ///
 /// Returns the number of bytes written (decompressed) into `output`.
-#[inline]
-fn decompress_internal<const USE_DICT: bool>(
+#[inline(always)] // (always) necessary to get the best performance in non LTO builds
```

Comment:
do you remember how much this was? inline(always) are a little risky that high, since the underlying inlines can easily change

Reply:
10% for the 34kb dataset. To reproduce just turn off LTO for in the cargo.toml file

Reply:
I should also mention that this applies for the code in master as well. It's sort of unrelated to the PR title.